Decisions.TruCap/Steps/AuthSteps.cs

```diff
@@ -38,44 +39,75 @@ public class AuthSteps
 
             try
             {
-                HttpResponseMessage response = await client.SendAsync(request);
+                HttpResponseMessage response = client.Send(request);
                 response.EnsureSuccessStatusCode();
 
-                return LoginResponse.JsonDeserialize(await response.Content.ReadAsStringAsync());
+                Task<string> resultTask = response.Content.ReadAsStringAsync();
+                resultTask.Wait();
+                return LoginResponse.JsonDeserialize(resultTask.Result);
             }
-            catch (Exception ex)
+            catch (BusinessRuleException ex)
             {
-                if (ex.Message.Contains("timed out"))
-                {
-                    throw new Exception("TruCap+ took too long to respond and has timed out.", ex);
-                }
-
-                throw;
+                throw new BusinessRuleException("The request to TruCap+ was unsuccessful.", ex);
             }
         }
 
         public bool IsLoggedIn(TruCapAuthentication authentication,
             [PropertyClassification(0, "Override Base URL", "Settings")] string? overrideBaseUrl)
         {
             string baseUrl = ModuleSettingsAccessor<TruCapSettings>.GetSettings().GetBaseUrl(overrideBaseUrl);
-            Task<HttpResponseMessage> response = TruCapRest.TruCapGet(
-                $"{baseUrl}/login/status?sid={authentication.sid}&token={authentication.token}", authentication);
+            HttpClient client = new HttpClient();
+        
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, 
+                $"{baseUrl}/login/status?sid={authentication.sid}&token={authentication.token}");
+            request.Headers.Add("sid", authentication.sid);
+            request.Headers.Add("Authorization", $"Bearer {authentication.token}");
+        
+            try
+            {
+                HttpResponseMessage response = client.Send(request);
+                response.EnsureSuccessStatusCode();
+
+                Task<string> resultTask = response.Content.ReadAsStringAsync();
+                resultTask.Wait();
 
-            return response.Result.IsSuccessStatusCode;
+                return response.IsSuccessStatusCode;
+            }
+            catch (Exception ex)
+            {
+                throw new BusinessRuleException("The request to TruCap+ was unsuccessful.", ex);
```

Comment:
Since this is related to authentication, we should be more specific in the message to the user.